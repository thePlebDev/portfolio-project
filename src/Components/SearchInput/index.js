import React from'react'
import styled from 'styled-components'
import HighlightOffIcon from '@material-ui/icons/HighlightOff';

const Label = styled.label`
  display:flex;
  position:relative;
`

const Input = styled.input`
  width:100%;
  padding:10px;
  margin:0 auto;
  font-size:1.5rem;
`



const SearchInput=({state,handleChange,setShow})=>{
  return(
    <Label >
      <Input type="text" placeholder='Search blog posts...' onChange={(e)=>{handleChange(e)}} value={state} name="search" />
      <HighlightOffIcon style={{position:'absolute',top:'25%',right:'5%','z-index':999,cursor:'pointer'}} onClick={()=>setShow(false)}/>
    </Label>
  )
}

export default SearchInput
