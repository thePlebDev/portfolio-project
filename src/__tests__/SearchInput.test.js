import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import SearchInput from '../Components/SearchInput'



describe('testing SearchInput component',()=>{

  it('just testing stuff',()=>{
    const tree = renderer.create(<SearchInput/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should render the important stuff',()=>{
    const wrapper = shallow(<SearchInput/>)
    expect(wrapper.find('[data-testid="input"]').length).toBe(1)
    expect(wrapper.find('[data-testid="close"]').length).toBe(1)
  })
  it('should render the custom props',()=>{
    let state='state'
    const handleChange= jest.fn(()=>{})
    const setShow= jest.fn(()=>{})

    const wrapper = shallow(<SearchInput state={state} handleChange={handleChange} setShow={setShow}/>)
    expect(wrapper.find('[data-testid="input"]').props().value).toBe(state)

    wrapper.find('[data-testid="input"]').simulate('change',{ target:{name: 'search',value:'two pretty best friends'} })
    expect(handleChange.mock.calls.length).toBe(1)

    wrapper.find('[data-testid="close"]').simulate('click')
    expect(setShow.mock.calls.length).toBe(1)
  })
})
