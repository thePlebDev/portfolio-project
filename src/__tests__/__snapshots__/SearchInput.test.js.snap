// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SearchInput component just testing stuff 1`] = `
<label
  className="sc-bdnylx gJMbZK"
>
  <input
    className="sc-gtssRu ubynD"
    data-testid="input"
    name="search"
    onChange={[Function]}
    placeholder="Search blog posts..."
    type="text"
    value="Enter a state prop"
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    data-testid="close"
    focusable="false"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "position": "absolute",
        "right": "5%",
        "top": "25%",
        "z-index": 999,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
    />
  </svg>
</label>
`;
